common/buildcraft/silicon/tile/TileLaser.java

```diff
@@ -129,41 +144,46 @@ public long getMaxPowerPerTick() {
 
     @Override
     public void update() {
-        if (world.isRemote) {
-            return;
-        }
-        avgPower.tick();
         ticks++;
+        if (world.isRemote) {
+            // set laser render position on client side
+            if (ticks % (5 + world.rand.nextInt(10)) == 0 || targetPos == null) {
+                updateLaser();
+            }
+        } else {
```

Comment:
Please keep the old ````return```` statement rather than this else. Unless you have a reason for it?

Reply:
I just thought the if/else structure was cleaner for showing the client/server side updates. I can switch it back to having the old return if that's the pattern the buildcraft code uses.

Reply:
I mostly use
````java
 if (client) {
    // Do client tasks
    return;
}
````
as it makes all of the next lines less indented than otherwise (most of the time this doesn't matter, but sometimes it allows much longer lines to fit)